### PR TITLE
Add and delete node command for automate HA

### PR DESCRIPTION
**Pull request by vivekshankar1**
### :nut_and_bolt: Description: What code changed, and why?

Created command to add new node in chef automate HA, example: chef-automate node add --automate 10.0.1.2
Created command to delete node from chef automate HA, example: chef-automate node remove --automate 10.0.1.2

### :chains: Related Resources

https://chefio.atlassian.net/browse/SHIELD-171
https://chefio.atlassian.net/browse/SHIELD-172

### :+1: Definition of Done

Running the add node command will add node to HA setup
Running the remove node commands will remove node from HA setup.
You can verify this by running `chef-automate status`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2022-11-25 at 4 11 47 PM

Screenshot 2022-11-25 at 4 30 53 PM

Screenshot 2022-11-25 at 4 11 18 PM

https://user-images.githubusercontent.com/59958706/203970206-d1aa40a6-37e0-4347-8c1a-4f7122cb9264.mp4

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/haNodeAdd_test.go [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+Incorrect Chef-Server IP address format for ip ewewedw
+Incorrect OpenSearch IP address format for ip ewewedw
+Incorrect Postgresql IP address format for ip 10.2.1.657
+Incorrect Postgresql IP address format for ip ewewedw`)
+}
+
+func Test_addnode_readfile_error(t *testing.T) {
+	w := majorupgrade_utils.NewCustomWriterWithInputs("x")
+	flags := AddDeleteNodeHACmdFlags{
+		automateIp: "10.2.1.67",
```
Comment (atultherajput) [inline components/automate-cli/cmd/chef-automate/haNodeAdd_test.go:103]: Replace all hardcoded IP addresses with one of the below suggestions:

- Loopback addresses 127.0.0.0/8 in CIDR notation (from 127.0.0.0 to 127.255.255.255)
- Broadcast address 255.255.255.255
- Non-routable address 0.0.0.0
- Strings of the form 2.5.<number>.<number> as they [often match Object Identifiers](http://www.oid-info.com/introduction.htm) (OID)
- Addresses in the ranges 192.0.2.0/24, 198.51.100.0/24, 203.0.113.0/24, reserved for documentation purposes by [RFC 5737](https://datatracker.ietf.org/doc/html/rfc5737)
- Addresses in the range 2001:db8::/32, reserved for documentation purposes by [RFC 3849](https://datatracker.ietf.org/doc/html/rfc3849)

File: components/automate-cli/cmd/chef-automate/automateHADeployExistingInfra.go [hunk 2]
```diff
@@ -247,24 +247,15 @@ func (e *existingInfra) validateExternalDbFields() *list.List {
...
 	return ips
 }
 
-func extractIPsFromCertsByIPOpensearch(certsByIp []struct {
-	IP         string `toml:"ip"`
-	PrivateKey string `toml:"private_key"`
-	PublicKey  string `toml:"public_key"`
-	NodesDn    string `toml:"nodes_dn"`
-}) []string {
+func extractIPsFromCertsByIPOpensearch(certsByIp []CertByIP) []string {
```
Comment (atultherajput) [inline components/automate-cli/cmd/chef-automate/automateHADeployExistingInfra.go:258]: Both extractIPsFromCertsByIPOpensearch and extractIPsFromCertsByIP are similar. Can we replace one with another?

**Review by atultherajput — COMMENTED** [review]
Fix Code Smell and Security Hotspot raised by Sonarcloud.

- https://sonarcloud.io/project/security_hotspots?id=chef_automate&pullRequest=7563&resolved=false&types=SECURITY_HOTSPOT

- https://sonarcloud.io/project/issues?id=chef_automate&pullRequest=7563&resolved=false&types=CODE_SMELL

**Review by dishanktiwari2501 — APPROVED** [review]
Reviewed and Updated the doc accordingly.

**Review by atultherajput — APPROVED** [review]

**Review by vivek-yadav — APPROVED** [review]

**Review by jayvikramsharma1 — APPROVED** [review]

**Pull request merged**